test/services/session.service.spec.unit.ts

```diff
@@ -0,0 +1,13 @@
+import Session from "../../src/session/session";
+import * as sessionService from "../../src/services/session.service";
+import * as keys from "../../src/session/keys";
+
+describe ("cache service tests", () => {
+
+  it("should create a new empty promise to file session", async () => {
+    const session: Session = Session.newInstance();
+    await sessionService.createPromiseToFileSession(session);
+    await sessionService.updatePromiseToFileSessionValue(session, keys.COMPANY_NUMBER, "00006400");
+    expect(session.data.ptf_session.company_number).toEqual("00006400");
+  });
```

Comment:
needs a test to check it doesn't create a new ptf session if one exists

Reply:
Testing the midleware here is tricky as it involves explicitly getting the mocked ch session. 

Attempted to write a unit test earlier but there currently seems to be no handle on this ch session. The code does not seemed to be designed for testing this directly. 

Suggest covering this further down the line when we have completed the comapny number and check company pages along the lines of what is in the index.spec.unit tests for the authentication mdidleware - these provide a more indirect way of testing middleware.